### DISCUSSION TITLE
add markdown toolbar

### DISCUSSION THREAD

**Discussion by madeindjs** [GitJournal/GitJournal#802] [Feature Requests]
It maybe great if editor includes a markdown toolbar. This toolbar allow user to transform a text into bold, italic, etc..

You can check Markor example or Github toolbar bellow:

![Screenshot_20200602-213836](https://user-images.githubusercontent.com/11815139/83562460-e8663800-a519-11ea-81e3-97e15ccab173.jpg)

---
> Please vote on this issue with a 👍 reaction. It's one of the primary mechanisms used to prioritize issues. Please also subscribe to the issue if you're interested, as that way I can discuss possible implementations and share mockups.

**Comment by mipro98** [comment]
There might also be the possibility to add custom styles in the context menu on text selection: https://medium.com/androiddevelopers/custom-text-selection-actions-with-action-process-text-191f792d2999.

I've seen Telegram doing that (screenshot). This might not be as convenient to use but would also not take up that much space and be (maybe) easier to implement. 

![telegram](https://user-images.githubusercontent.com/28269133/83620630-19c41f80-a57d-11ea-8c71-663f1220b6e7.png)

**Comment by vHanda** [comment]
Version 1.74 has an experimental version of the markdown toolbar which can be enabled from Settings -> Experimental Features. I would love some feedback on it.

**Comment by alensiljak** [comment]
The toolbar idea is fantastic. It is often complicated to input some of the syntax on mobile keyboards, like square brackets and indentation. I would recommend adding the syntax which is quite hard to write, like:

- links, 
- images, 
- wiki-links
- indentation (+/-)

Other than that, I can recommend trying out the Open Source keyboard called FlorisBoard. It is in alpha state but already very functional and practical for entering special characters due to the lack of word correction and the built-in toolbar with the cursor control and clipboard operations. 
Also, see the scrolling Markdown toolbar in Markor. That has literally everything and the kitchen sink in the toolbar, which is quite handy for beginners.
Cheers and thanks a million for this inspiring app!
